File: app/src/main/java/com/sdwfqin/quickseed/ui/mvvm/WeatherMvvmActivity.java
package com.sdwfqin.quickseed.ui.mvvm;

import androidx.lifecycle.ViewModelProvider;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.blankj.utilcode.util.LogUtils;
import com.sdwfqin.quickseed.base.ArouterConstants;
import com.sdwfqin.quickseed.base.SampleBaseMvvmActivity;
import com.sdwfqin.quickseed.databinding.ActivityWeatherMvvmBinding;

/**
 * 获取天气信息 mvvm
 * <p>
 *
 * @author 张钦
 * @date 2020/4/14
 */
@Route(path = ArouterConstants.COMPONENTS_MVVM)
public class WeatherMvvmActivity extends SampleBaseMvvmActivity<ActivityWeatherMvvmBinding, WeatherViewModel> {

    @Override
    protected ActivityWeatherMvvmBinding getViewBinding() {
        return ActivityWeatherMvvmBinding.inflate(getLayoutInflater());
    }

    @Override
    protected WeatherViewModel getViewModel() {
        return new ViewModelProvider(this).get(WeatherViewModel.class);
    }

    @Override
    protected void initEventAndData() {
        mTopBar.setTitle("MVVM DEMO");
        mTopBar.addLeftBackImageButton().setOnClickListener(v -> finish());

        mBinding.setViewModel(mVm);
        mBinding.setHandlers(new WeatherMvvmHandlers());
        mBinding.setLifecycleOwner(this);

        mVm.birthDate.observe(this, LogUtils::e);

        mVm.loadWeather();
    }

    @Override
    protected void commonNetworkErrorListener(Throwable throwable) {
        showMsg(throwable.getMessage());
    }

}

File: app-kt/src/main/java/io/github/sdwfqin/app_kt/ui/mvvm/WeatherMvvmActivity.kt
package io.github.sdwfqin.app_kt.ui.mvvm

import androidx.lifecycle.ViewModelProvider
import com.alibaba.android.arouter.facade.annotation.Route
import io.github.sdwfqin.app_kt.constants.ArouterConstants
import io.github.sdwfqin.app_kt.databinding.ActivityWeatherMvvmBinding
import io.github.sdwfqin.samplecommonlibrary.base.SampleBaseMvvmActivity

/**
 * 获取天气信息 mvvm
 *
 * @author 张钦
 * @date 2020/4/14
 */
@Route(path = ArouterConstants.COMPONENTS_MVVM)
class WeatherMvvmActivity : SampleBaseMvvmActivity<ActivityWeatherMvvmBinding, WeatherViewModel>() {
    override fun getViewBinding(): ActivityWeatherMvvmBinding {
        return ActivityWeatherMvvmBinding.inflate(layoutInflater)
    }

    override fun getViewModel(): WeatherViewModel {
        return ViewModelProvider(this).get(WeatherViewModel::class.java)
    }

    override fun initEventAndData() {
        mTopBar.setTitle("MVVM DEMO")
        mTopBar.addLeftBackImageButton().setOnClickListener { finish() }
        mBinding.viewModel = mVm
        mBinding.lifecycleOwner = this
        mVm.loadWeather()
    }

    override fun commonNetworkErrorListener(throwable: Throwable) {
        showMsg(throwable.message)
    }
}